packages/vx-xychart/src/classes/DataRegistry.ts

```diff
@@ -0,0 +1,54 @@
+import { AxisScaleOutput } from '@vx/axis';
+import { ScaleConfig } from '@vx/scale';
+import { DataRegistryEntry } from '../types/data';
+
+/** A class for holding data entries */
+export default class DataRegistry<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  YScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  Datum = unknown,
+  DataKey extends string = string
+> {
+  private registry = new Map<
```

Comment:
if key is only `string`, plain map `{ [key: string]: DataRegistryEntry }` is more lightweight than `Map` type.